src/Android/Xamarin.Android/Samples/Augmented Reality/NavigateAR/RouteViewerAR.cs

```diff
@@ -0,0 +1,293 @@
+            SetRoute(PassedRouteResult.Routes[0]);
+        }
+
+        private void SetRoute(Route inputRoute)
+        {
+            // Clear any existing route graphics.
+            _routeOverlay.Graphics.Clear();
+
+            // Create a new route graphic and show it in the view.
+            Graphic routeGraphic = new Graphic(inputRoute.RouteGeometry);
+            _routeOverlay.Graphics.Add(routeGraphic);
+
+            // Configure the overlay to show routes 3 meters above the ground.
+            _routeOverlay.SceneProperties.SurfacePlacement = SurfacePlacement.Relative;
+            _routeOverlay.SceneProperties.AltitudeOffset = 3;
+        }
+
+        private void StartTurnByTurn()
+        {
+            // Create the route tracker.
+            _routeTracker = new RouteTracker(PassedRouteResult, 0);
+
+            // Create the speech synthesizer.
+            _textToSpeech = new TextToSpeech(this, null, "com.google.android.tts");
+
+            // Listen for updated guidance.
+            _routeTracker.NewVoiceGuidance += RouteTracker_VoiceGuidanceChanged;
+            _routeTracker.TrackingStatusChanged += RouteTracker_TrackingStatusChanged;
+
+            // Prevent re-navigating.
+            _navigateButton.Enabled = false;
+        }
+
+        private void RouteTracker_TrackingStatusChanged(object sender, RouteTrackerTrackingStatusChangedEventArgs e)
+        {
+            RunOnUiThread(() =>
+            {
+                _helpLabel.Text = _routeTracker.GenerateVoiceGuidance().Text;
+            });
+        }
+
+        private void RouteTracker_VoiceGuidanceChanged(object sender, RouteTrackerNewVoiceGuidanceEventArgs e)
+        {
+            RunOnUiThread(() =>
+            {
+                _helpLabel.Text = e.VoiceGuidance.Text;
+
+                _textToSpeech.Stop();
+                _textToSpeech.Speak(e.VoiceGuidance.Text, QueueMode.Flush, null, null);
+            });
+        }
+
+        private void ShowMessage(string message, string title, bool closeApp = false)
+        {
+            // Show a message and then exit after if needed.
+            var dialog = new AlertDialog.Builder(this).SetMessage(message).SetTitle(title).Create();
+            if (closeApp)
+            {
+                dialog.SetButton("OK", (o, e) =>
+                {
+                    Finish();
+                });
+            }
+            dialog.Show();
+        }
+
+        protected override async void OnPause()
+        {
+            base.OnPause();
+            await _arSceneView.StopTrackingAsync();
+        }
+
+        protected override async void OnResume()
+        {
+            base.OnResume();
+
+            // Start AR tracking without location updates.
+            await _arSceneView.StartTrackingAsync(ARLocationTrackingMode.Continuous);
+        }
+
+        protected override void OnDestroy()
+        {
+            base.OnDestroy();
+            if (_routeTracker != null)
```

Comment:
Do you need to call `await _arSceneView.StopTrackingAsync();` here?

Reply:
I don't think so since it would have been called in OnPause.